model/lollipop_keys.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  CosmosErrors,
+  toCosmosErrorResponse
+} from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { flow, pipe } from "fp-ts/lib/function";
+import { enumType } from "@pagopa/ts-commons/lib/types";
+
+const LOLLIPOPKEYS_MODEL_PK_FIELD = "assertionRef" as const;
+const LOLLIPOPKEYS_MODEL_ID_FIELD = LOLLIPOPKEYS_MODEL_PK_FIELD;
+
+export const Ttl = t.interface({
+  ttl: NonNegativeInteger // do we need this to be -1 in some cases?
+});
+export type Ttl = t.TypeOf<typeof Ttl>;
+
+// The time for which we want to reserve a key during login process (in seconds)
+export const TTL_VALUE_FOR_RESERVATION = (60 * 15) as NonNegativeInteger; // 15m
+// The time for which we want to keep the popDocument
+export const TTL_VALUE_AFTER_UPDATE = (60 *
+  60 *
+  24 *
+  365 *
+  2) as NonNegativeInteger; // 2y
+
+export enum PopDocumentStatusEnum {
+  "PENDING" = "PENDING",
+  "VALID" = "VALID",
+  "REVOKED" = "REVOKED"
+}
+export const PopDocumentStatus = enumType<PopDocumentStatusEnum>(
+  PopDocumentStatusEnum,
+  "PopDocumentStatus"
+);
+
+// fiscal code - AssertionRefsha256 | AssertionRefSha384 | AssertionRefSha512
+export const AssertionFileName = PatternString(
+  "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]-(sha256-[A-Za-z0-9-_=]{1,44}|sha384-[A-Za-z0-9-_=]{1,66}|sha512-[A-Za-z0-9-_=]{1,88})$"
+);
+export type AssertionFileName = t.TypeOf<typeof AssertionFileName>;
+
+export const PendingPopDocument = t.interface({
+  assertionRef: AssertionRef,
+  pubKey: NonEmptyString,
+  status: t.literal(PopDocumentStatusEnum.PENDING)
+});
+export type PendingPopDocument = t.TypeOf<typeof PendingPopDocument>;
+
+export const NotPendingPopDocument = t.interface({
+  assertionFileName: AssertionFileName,
+  assertionRef: AssertionRef,
+  assertionType: AssertionType,
+  expiredAt: Timestamp,
+  fiscalCode: FiscalCode,
+  pubKey: NonEmptyString,
+  status: t.union([
+    t.literal(PopDocumentStatusEnum.VALID),
+    t.literal(PopDocumentStatusEnum.REVOKED)
+  ])
+});
+export type NotPendingPopDocument = t.TypeOf<typeof NotPendingPopDocument>;
+
+export const PopDocumentBase = t.interface({
+  assertionRef: AssertionRef,
+  pubKey: NonEmptyString
+});
+export type PopDocumentBase = t.TypeOf<typeof PopDocumentBase>;
+
+// T type
+export const PopDocument = t.union([NotPendingPopDocument, PendingPopDocument]);
+export type PopDocument = t.TypeOf<typeof PopDocument>;
+
+// TN type
+export const NewPopDocument = t.intersection([PopDocument, Ttl]);
+export type NewPopDocument = t.TypeOf<typeof NewPopDocument>;
+
+// TR type
+export const RetrievedPopDocument = t.intersection([
+  PopDocument,
+  RetrievedVersionedModelTTL
+]);
+export type RetrievedPopDocument = t.TypeOf<typeof RetrievedPopDocument>;
+
+export class LolliPOPKeysModel extends CosmosdbModelVersionedTTL<
+  PopDocument,
+  NewPopDocument,
+  RetrievedPopDocument,
+  typeof LOLLIPOPKEYS_MODEL_ID_FIELD
```

Comment:
We should add both model id and pk field names.
```suggestion
  typeof LOLLIPOPKEYS_MODEL_ID_FIELD,
  typeof LOLLIPOPKEYS_MODEL_PK_FIELD
```

Reply:
I thought we wanted to use the same field for modelId and pk

Reply:
in the final JSON object that we send to cosmosdb there will be:
```
  modelId: <assertionRef>-<version>
  partitionKey: <assertionRef>
```